src/openapi.yaml

```diff
@@ -7149,6 +7141,29 @@ components:
       - price
       - url
       - uuid
+    FormRegistrationBackend:
+      type: object
+      properties:
+        key:
+          type: string
+          description: The key to use to refer to this configuration in form logic.
+          maxLength: 50
+        name:
+          type: string
+          description: A recognisable name for this backend configuration.
+          maxLength: 255
+        backend:
+          type: string
+          title: Registration backend
+          maxLength: 100
+        options:
+          type: object
+          additionalProperties: {}
+          title: Registration backend options
```

Comment:
That was a bit of a lie:

 - There are no plugins with `allow_null` `configuration_options`
 - `not plugin.configuration_options` in `validate_backend_options` is hard to be true because
```python
    def check_plugin(self, plugin):
        if not plugin.configuration_options:
            raise ValueError(...
```

But yes, it now handles it and tests for it.

Reply:
`plugin.configuration_options` in this case expects a `serializers.Serializer` subclass, I don't think it says anything about that serializer being `allow_null` :thinking: 

So checking things - the serializer points to the model field, which has `allow_null`, so DRF treats this as a nullable JSONField (which explains the old schema). Next, the serializer validation runs:

```py
self.validate_backend_options(
            attrs,
            "registration_backend",
            "registration_backend_options",
            registration_register,
        )
```
which results in the actual options validation via:

```py
serializer = plugin.configuration_options(data=options)
```
there are code paths here where `options is None`, but since there is no `allow_null=True` kwarg, I think it was effectively indeed blocked and would result in validation errors.